features/machine/machine.feature

```diff
@@ -141,4 +141,101 @@ Feature: Machine features testing
     Given I scale the machineset to +2
     Then the machineset should have expected number of running machines
 
-    
+  # @author zhsun@redhat.com
+  @admin
+  @destructive
+  Scenario Outline: Machineset should have relevant annotations to support scale from/to zero
+    Given I have an IPI deployment
+    And I switch to cluster admin pseudo user
+    And I use the "openshift-machine-api" project
+    And I pick a random machineset to scale
+
+    Given I run the :get admin command with:
+      | resource      | machineset              |
+      | resource_name | <%= machine_set.name %> |
+      | o             | yaml                    |
+    Then the step should succeed
+    And I save the output to file> machineset.yaml
+
+    # Create a machineset with a valid instanceType
+    And I replace content in "machineset.yaml":
+      | <%= machine_set.name %> | machineset-valid |
+      | <re_type_field>         | <valid_value>    |
+      | /replicas:.*/           | replicas: 1      |
+
+    When I run the :create admin command with:
+      | f | machineset.yaml |
+    Then the step should succeed
+    And admin ensures "machineset-valid" machineset is deleted after scenario
+
+    When I run the :annotate admin command with:
+      | resource     | machineset        |
+      | resourcename | machineset-valid  |
+      | overwrite    | true              |
+      | keyval       | new=new           |
+    Then the step should succeed
+
+    When I run the :describe admin command with:
+      | resource | machineset      |
+      | name     |machineset-valid |
+    Then the step should succeed
+    And the output should contain:
+      | machine.openshift.io/memoryMb: |
+      | machine.openshift.io/vCPU:     |
```

Comment:
The provider specific annotations will be added to machineset according to instance type so that autoscaler can scale from zero using these annotations.

Reply:
Thanks